zipkin-server/src/main/java/zipkin/server/ZipkinServerConfiguration.java

```diff
@@ -46,8 +48,14 @@ CollectorSampler traceIdSampler(@Value("${zipkin.collector.sample-rate:1.0}") fl
 
   @Bean
   @ConditionalOnMissingBean(CollectorMetrics.class)
-  CollectorMetrics metrics(CounterService counterService, GaugeService gaugeService) {
-    return new ActuateCollectorMetrics(counterService, gaugeService);
+  CollectorMetrics metrics(Optional<CounterBuffers> counterBuffers, Optional<GaugeBuffers> gaugeBuffers) {
```

Comment:
I'd probably make a test case for this class, particularly for those like me who had no idea you could inject an optional :P